File: app/src/main/java/msr/mirudl/app/MiruDLApp.java
package msr.mirudl.app;

import android.app.Application;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.os.Build;

import msr.mirudl.shared.download.AndroidFileStorage;

public class MiruDLApp extends Application {
    public static final String CHANNEL_DOWNLOADS = "mirudl_downloads";

    @Override
    public void onCreate() {
        super.onCreate();
        AndroidFileStorage.init(this);
        CrashLogger.init(this);
        createNotificationChannels();
    }

    private void createNotificationChannels() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            NotificationChannel channel = new NotificationChannel(
                    CHANNEL_DOWNLOADS,
                    "Downloads",
                    NotificationManager.IMPORTANCE_LOW
            );
            channel.setDescription("Download progress notifications");
            NotificationManager mgr = getSystemService(NotificationManager.class);
            if (mgr != null) mgr.createNotificationChannel(channel);
        }
    }
}

File: app/src/main/java/msr/mirudl/app/MiruDLApp.kt
package msr.mirudl.app

import android.app.Application
import android.app.NotificationChannel
import android.app.NotificationManager
import android.os.Build
import msr.mirudl.shared.download.AndroidFileStorage

class MiruDLApp : Application() {

    override fun onCreate() {
        super.onCreate()
        AndroidFileStorage.init(this)
        CrashLogger.init(this)
        createNotificationChannels()
    }

    private fun createNotificationChannels() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            val channel = NotificationChannel(
                CHANNEL_DOWNLOADS,
                "Downloads",
                NotificationManager.IMPORTANCE_LOW
            ).apply {
                description = "Download progress notifications"
            }
            val mgr = getSystemService(NotificationManager::class.java)
            mgr?.createNotificationChannel(channel)
        }
    }

    companion object {
        const val CHANNEL_DOWNLOADS = "mirudl_downloads"
    }
}
